[REL] Migration account_invoice_move_currency to 11.0

nicomacr opened the pull request:
[IMP] to set secundary currency with a boolean
[FIX] readme
[MOV] change the xml files path accordin guidelines

nicomacr committed:
[REL] Migration account_invoice_move_currency to 11.0
[IMP] to set secundary currency with a boolean
[FIX] readme
[MOV] change the xml files path accordin guidelines

nicomacr committed:
[ADD] add enterprice-extention to oca_dependecies

ValentinoDefelice reviewed (commented):
Hi Nico! I was looking how to configure this module inside Odoo and i havent found this information in README. Thank you so much.

zaoral commented on account_invoice_move_currency/__manifest__.py:
```diff
@@ -19,21 +19,22 @@
-    'depends': ['account'],
-    'test': [],
+    'depends': [
+        'account_ux'
```
Why account_ux and not account dependency as it was before?

nicomacr replied:
@zaoral because the functionality improvement to change invoice currency is in account ux.

zaoral commented on account_invoice_move_currency/__manifest__.py:
```diff
@@ -19,21 +19,22 @@
+        'wizards/account_change_currency_views.xml',
     ],
     'demo': [
         # 'demo/account_invoice_demo.yml',
```
If this demo is not used then we could remove it from here and also delete the file

zaoral commented on account_invoice_move_currency/README.rst:
```diff
@@ -1,5 +1,13 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.png
+   :target: https://www.gnu.org/licenses/agpl
    :alt: License: AGPL-3
 
```
The module description is confusing, please improve this part

zaoral reviewed (changes requested):
Hi @nicomacr
Here is the review

First need to tell you that I  was not able to make functional review this module in runbot, this because it seems to be a problem to have enterprise-extensions now as part of the requirements of this repo.
@jjscarafia what do you think we could do here?

Second here are some technical review, in order to make the improvements:

- ``finalize_invoice_move_lines`` have a ``ValidationError`` exception. this one should be ``UserError`` type instead.
- ``# TODO implement this`` review if this one will be applied in this version.
- ``_compute_residual`` compute method:

   -  remove the api.one
    - adapt method to be multi-record-aware
    - IMHO will be nice to remove the iteration ``for line in self.sudo().move_id.line_ids ... 
 if line.account_id.internal_type in ('receivable', 'payable'):`` and use a recorset.filtered() directly instead. 
   - a multiple direct assignation is made, could you try to use update() instead to see if it works here

nicomacr committed:
[FIX] update README

nicomacr committed:
[FIX] fix demo data

nicomacr committed:
[FIX] remove the api.one
[IMP] adapt method to be multi-record-aware
[FIX] _compute_residual, remove api.one decorator

nicomacr committed:
[FIX] some fixes
[REM] comment unsed lines

zaoral commented on account_invoice_move_currency/models/account_invoice.py:
```diff
@@ -45,27 +44,32 @@ def _compute_residual(self):
-            return super(AccountInvoice, self)._compute_residual()
+        for rec in self:
+            if not rec.move_currency_id:
+                return super(AccountInvoice, self)._compute_residual()
```
should be a super without return
and add a new line after that one with continue statement.
This in order to be able to properly compute the value for each value

zaoral commented on account_invoice_move_currency/README.rst:
```diff
@@ -15,25 +15,32 @@ Account Invoice Move Currency
 
 This module extends the functionality of account invoice to
-allows force the seat on an invoice generated save the data in another currency.
+allows force the seat on an invoice generated save the data in another currency. This module adds funcionality to save the secondary currency in the wizard "change" that introduce the module "Account Usability"
```
* "the seat" is bad english to refer to "el asiento", should be journal.entries
- "generated save the data", I don not understand what it means here we have save verb without sustantive.
- account usability is accout ux.

zaoral commented on account_invoice_move_currency/README.rst:
```diff
@@ -15,25 +15,32 @@ Account Invoice Move Currency
 To use this module, you need to:
 
+#. Go to an invoice and use the button "change"
+#. If you select an currency to and it's the same as company currency, it's able to set to save the actual currency into "Secondary Currency"
```
misspelling error

- "a currency" instead of "an currency" -Use of currency to,  this is like technical . something like:

If you have an invoice which currency is different from the company's currency, and you choose to change it to the company currency, you are able to record the information of the first one (both currency and rate) in order to be able to know the equivalent value in the foreign  currency as a informative data.

nicomacr committed:
[FIX] update readme
[FIX] some fix to performance in compute fields

nicomacr committed:
[FIX] oca dependencies

zaoral reviewed (changes requested):
@nicomacr 
Good morning Nico

Functional review +1
About technical I have just saw some rst errors. Please apply this changes in order to approve and merge. 

![image](https://user-images.githubusercontent.com/7593953/40056995-694982f6-5823-11e8-897a-f6ce399dc744.png)

Best Regards

nicomacr committed:
[FIX] Update Readme

jjscarafia committed:
Merge branch '11.0' into 11.0-t-9492-nmr
